File: java_pro/src/main/java/jv/com/tencent/cain/design_pattern/decorator/Person.java
package jv.com.tencent.cain.design_pattern.decorator;

/**
 * @author cainjiang
 * @date 2019-03-31
 * 被装饰者
 */
public class Person {
    private String name;

    public Person() {
    }

    public Person(String name) {
        this.name = name;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    protected void show() {
        System.out.println("装扮的" + name);
    }
}

File: java_pro/src/main/kotlin/kt/com/tencent/cain/desigin_pattern/decorator/Person.kt
package kt.com.tencent.cain.desigin_pattern.decorator

/**
 * @author cainjiang
 * @date 2019-03-31
 * 被装饰者
 */
open class Person(val name: String = "程序员") {
    open fun show() {
        println("装扮的$name")
    }
}